### PR TITLE
[feat] check for updates

### LINKED ISSUE #223: Print a message if a newer version is available
Hi 👋

I found this feature very useful in the [GitHub CLI application](https://github.com/cli/cli/blob/trunk/internal/update/update.go) and [npm](https://github.com/npm/cli/blob/latest/lib/utils/update-notifier.js).

I think it'll be useful for users who install [comtrya](https://github.com/comtrya/comtrya) using the installation script.
And this will let them know that a new version is available and it's time to update.

Ready-made crates to do that:
* https://github.com/mgrachev/update-informer
* https://github.com/tarikeshaq/update-notifier

### PR DESCRIPTION
**Pull request by mgrachev**
## I'm submitting a

- [ ] bug fix
- [x] feature
- [ ] documentation addition

Close #223

### CODE REVIEW

**Review by martintc — COMMENTED** [review]
As it is, it looks good to me code wise. The only thing I can that we may want to change is where the check for update happens in execution. I will see what Rawkode and Icepuma thing.

**Review by martintc — APPROVED** [review]

**Comment by icepuma** [comment]
Nice addition. I think the integration should be more subtle. Maybe display the new version with the `comtrya version` command.

**Comment by icepuma** [comment]
We talked a bit at our [Discord](https://rawkode.chat/) about the feature. Could you please add a config switch to `Comtrya.yaml` with the value `disableUpdateCheck: <true|false>` to opt-out of the update check.

**Comment by icepuma** [comment]
If you want to iterate a bit more over the feature, you could convert the PR to a draft.

**Comment by mgrachev** [comment]
> We talked a bit at our [Discord](https://rawkode.chat/) about the feature. Could you please add a config switch to `Comtrya.yaml` with the value `disableUpdateCheck: <true|false>` to opt-out of the update check.

Done.

File: app/src/main.rs [hunk 1]
```diff
@@ -80,12 +80,52 @@ fn main(args: GlobalArgs) -> anyhow::Result<()> {
...
     let runtime = Runtime {
         args,
         config,
         contexts,
     };
 
-    execute(runtime)
+    execute(runtime)?;
+
+    if can_check_for_updates {
```
Comment (icepuma) [inline app/src/main.rs:94]: I would put this in right before the `execute` - block.

Comment (mgrachev) [reply app/src/main.rs]: 👌

**Comment by rawkode** [comment]
Great addition. Thank you, @mgrachev

**Review by rawkode — APPROVED** [review]

**Review by icepuma — APPROVED** [review]

**Comment by icepuma** [comment]
I'll merge when everything is green

**Comment by icepuma** [comment]
macOS build fails, because it is deprecated. I'll fix it later

**Pull request merged**